frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx

```diff
@@ -266,11 +266,12 @@ function InsightCardInternal(
     }: InsightCardProps,
     ref: React.Ref<HTMLDivElement>
 ): JSX.Element {
+    const insight = getQueryBasedInsightModel(legacyInsight)
```

Comment:
We replace the insight's `filters` with a `query`, if that hasn't happened backend-side yet.